_posts/2017-11-16-totw-112.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Tip of the Week #112: emplace vs. push_back"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/112
+type: markdown
+order: "112"
+---
+
+Originally posted as totw/112 on 2016-02-25
+
+By Geoff Romer (gromer@google.com)
+
+Revised 2017-08-30
+
+*"The less we use our power, the greater it will be." — Thomas Jefferson*
+
+As you may already know (and if not, [TotW 65](/tips/65)), C++11 introduced a
+powerful new way to insert items into containers: `emplace` methods. These let
+you construct an object in-place inside the container, using any of the object's
+constructors. That includes the move and copy constructors, so it turns out any
+time you could use a `push` or `insert` method, you can use `emplace` instead,
```

Comment:
push() insert()

Since "emplace" refers to a family of methods: 
emplace -> an `emplace` method

Reply:
strictly, the same is true of push. (push_back(), push(), push_front()).

Reply:
yeah, I think these are better without the ()s since we aren't talking about a single API but a family of conceptually related / named APIs.

Reply:
would still say "an \`emplace\` method" however.

Reply:
done